src/ol/interaction/translate.js

```diff
@@ -234,6 +240,35 @@ ol.interaction.Translate.prototype.setHitTolerance = function(hitTolerance) {
 
 
 /**
+ * @inheritDoc
+ */
+ol.interaction.Translate.prototype.setMap = function(map) {
+  var oldMap = this.getMap();
+  ol.interaction.Pointer.prototype.setMap.call(this, map);
+  this.updateState_(oldMap);
+};
+
+
+/**
+ * @param {ol.Map} oldMap Old map.
+ * @private
+ */
+ol.interaction.Translate.prototype.updateState_ = function(oldMap) {
```

Comment:
My point is that you are registering `updateState_` as a [listener for object change events](https://github.com/tchandelle/openlayers/blob/d9976b990ef4a84766da166d87307919196befa8/src/ol/interaction/translate.js#L87-L89).  The first argument to event listeners is an event, in this case `ol.Obect.Event`.  But your function expects `ol.Map`.

The compiler isn't catching the type mismatch.  And the current code will not break, but if someone else modifies this function and expects `oldMap` to always be an `ol.Map`, they will be surprised when they receive `ol.Object.Event` instead.